src/Keyboard/index.tsx

```diff
@@ -0,0 +1,246 @@
+import Keyboard from 'react-simple-keyboard';
+import 'react-simple-keyboard/build/css/index.css';
+import './Keyboard.css';
+import { useEffect, useLayoutEffect, useRef, useState } from 'react';
+import { createPortal } from 'react-dom';
+import {DISPLAY_WIDTH, DISPLAY_HEIGHT } from '../const/display';
+
+interface OnScreenKeyboardProps {
+  /**
+   * The text that's currently being searched
+   */
+  searchQuery: string;
+  /**
+   * A handler to update the search text
+   */
+  searchUpdateHandler: (arg0: string) => void;
+  /**
+   * A handler to initiate a search
+   */
+  triggerSearchHandler: () => void;
+}
+
+
+/**
+* Enum for handling the casing of the keyboard
+*/
+enum CASE {
+  /** lowercase layout */
+  DEFAULT,
+  /** next character uppercase, then back to lowercase */
+  SHIFTED,
+  /** uppercase layout */
+  LOCKED,
+  /** next character lowercase, then back to uppercase */
+  UNSHIFTED,
+}
+
+const OnScreenKeyboard = ({
+  searchQuery,
+  searchUpdateHandler,
+  triggerSearchHandler,
+}: OnScreenKeyboardProps) => {
+  /**
+   * Whether the keyboard should be shown on screen.
+   */
+  const [isVisible, setVisible] = useState(false);
+
+  /**
+   * Set a ref to the modal for updating placement, etc.
+   */
+  const keyboardModalRef = useRef<HTMLDivElement>(null);
+  const backdropRef = useRef<HTMLDivElement>(null);
+  /**
+   * The X,Y coordinates on the page at which the center point of the keyboard
+   * should be rendered
+   */
+  const [coordinates, setCoordinates] = useState<[number, number]>([540, 960]);
+
+  /**
+   * Handle setting the event listener to open the keyboard when clicking on
+   * the body, then close it when clicking outside the modal
+   */
+  useEffect(() => {
+    if (isVisible) {
+      const background = backdropRef.current;
+      const closeKeyboard = (evt: MouseEvent) => {
+        if (evt.target === background) {
+          searchUpdateHandler('')
+          setVisible(false);
+        };
+      }
```

Comment:
I think there's an extra semi-colon line 70 and missing ones on 68 and 71.

Reply:
Fixed in 52a09ef